deps/ntlmclient/ntlm.c

```diff
@@ -260,6 +260,9 @@ static inline bool write_buf(
 	const unsigned char *buf,
 	size_t len)
 {
+	if (!len)
+		return true;
```

Comment:
Same as with the other PR, splitting he ntlmclient changes into a separate commit would make it easier for @ethomson to cherry-pick.